### DISCUSSION TITLE
Why does clear() not release memory?

### DISCUSSION THREAD

**Discussion by ghost** [doctrine/doctrine2#9477] [Support Questions]
```php
echo get_memory_usage(true); // x MB

// could also assigne to a variable and then unset
$repo->findBy([], [], 10);

echo get_memory_usage(true); // x+100 MB

$em->clear();

echo get_memory_usage(true); // still x+100 MB
```

Unit of work has been cleared. No more references to entities (right?). Tested also with env=prod. Can anybody help?
Memory release happens for other objects when using unset(), overriding variable that holds a reference or when exiting a scope.
Doctrine 2.5.12.

**Comment by SenseException** [comment]
`$em->clear();` doesn't automatically garbage-collect when invoked. Entity objects still can be available after `$em->clear();`, but they are detached, so they are still in memory.

BTW: Your ORM version is outdated and should be updated.

**Reply by CsabaNa** [reply]
I just updated my doctrine(to 2.12.3) and it seems it is true even I force php to free memory with `gc_collect_cycles` and `unset`.
the code is really simple just like the original:
```
echo 'start: ' . get_memory_usage(true) . "\n"; // start x MB
while ($list = $repo->findBy([], [], 10)) {
    echo 'cycle starts: ' . get_memory_usage(true) . "\n"; // x+cs MB
    unset($list);
    $em->clear();
    gc_collect_cycles();
    echo 'cycle ends: ' . get_memory_usage(true) . "\n"; // x+cs+ MB
}
```
even you did nothing memory usage still increasing just because you called `clear()` actually without it you have more memory left.

**Reply by beberlei** [reply]
Do you have any listeners attached to this entity that keep a reference of the entities?

**Reply by CsabaNa** [reply]
I have only one listener:
```
class DoctrineTrackingPolicyListener
{
    public function loadClassMetadata(LoadClassMetadataEventArgs $args): void
    {
        $classMetadata = $args->getClassMetadata();
        $classMetadata->setChangeTrackingPolicy(
            ClassMetadataInfo::CHANGETRACKING_DEFERRED_EXPLICIT
        );
    }
}

```

**Reply by CsabaNa** [reply]
The Entities are simple but do have JSON fields in them.
